modules/pipeline/pipengine/actionexecutor/plugins/apitest/apitest.go

```diff
@@ -57,7 +57,32 @@ func (d *define) Create(ctx context.Context, task *spec.PipelineTask) (interface
 }
 
 func (d *define) Start(ctx context.Context, task *spec.PipelineTask) (interface{}, error) {
-	logic.Do(ctx, task)
+	executorDoneCh := ctx.Value(spec.MakeTaskExecutorCtxKey(task)).(chan interface{})
+	if executorDoneCh == nil {
+		return nil, fmt.Errorf("wait: failed to get exector channel, pipelineID: %d, taskID: %d", task.PipelineID, task.ID)
+	}
+	go func() {
+		logic.Do(ctx, task)
+
+		latestTask, err := d.dbClient.GetPipelineTask(task.ID)
+		if err != nil {
+			fmt.Printf("failed to query latest task, err: %v \n", err)
```

Comment:
if get db error, this api-test task will never stop?

Reply:
The query error will not be running all the time. The latest status query here is just to terminate the operation early. This code can be removed because it will always be queried in the status method of the executer, but the scheduled time of the status is incremented. Reach 10s, if the asynchronous execution does not return early, the execution time will be longer